src/plugins/lua/reputation.lua

```diff
@@ -992,6 +995,19 @@ local function reputation_redis_init(rule, cfg, ev_base, worker)
   local i = 1
   {% for w in windows %}
     redis.call('HSET', KEYS[1], 'v' .. '{= w.name =}', scores[i])
+    local top_size_max = {= neg_top_max =}
+    local top_name = '{= neg_top =}'
+    if scores[i] >= 0 then
+      top_name = '{= pos_top =}'
+      top_size_max = {= pos_top_max =}
+    end
+    local top_size = redis.call('ZCARD', top_name)
+    if top_size > top_size_max then
+      redis.call('ZREMRANGEBYRANK', top_name, 0, -(top_size_max + 1))
+    end
+    redis.call('ZADD', KEYS[1] .. top_name, math.abs(scores[i]), '{= w.name =}')
```

Comment:
You add to list named `KEYS[1]..top_name` but clean and do all other stuff with `top_name`. It looks like a mistake to me.

I think we should add `KEYS[1]` to zlist named `top_name` with the appropriate score. If we have multiple windows it's a problem but from what I see that's the only way to achieve the goal.

Reply:
Oh, ok. I just thought that user will know which key is he working with. I'll change that.

Reply:
It might be better to keep it the way I'm doing it now, but also have two `zsets` (for negative and positive) `keys` and scores as the sum of all the window scores that are associated with this `key` that are stored in the `KEYS[1] .. top_name`. What do you think?